plugins/browser-plugin-media-tracking/src/index.ts

```diff
@@ -0,0 +1,230 @@
+import { MediaPlayerEvent } from './contexts';
+import { findMediaElem } from './findMediaElement';
+import { buildMediaEvent } from './buildMediaEvent';
+import { MediaProperty } from './mediaProperties';
+
+declare global {
+  interface HTMLVideoElement {
+    autoPictureInPicture: boolean;
+    disableRemotePlayback: boolean;
+    disablePictureInPicture: boolean;
+  }
+  interface HTMLAudioElement {
+    disableRemotePlayback: boolean;
+  }
+  interface Document {
+    pictureInPictureElement: Element;
+  }
+}
+
+const _trackers: Record<string, BrowserTracker> = {};
+
+export function MediaTrackingPlugin(): BrowserPlugin {
+  return {
+    activateBrowserPlugin: (tracker: BrowserTracker) => {
+      _trackers[tracker.id] = tracker;
+    },
+  };
+}
+
+const trackedIds: Record<string, trackedElement> = {};
+
+export function enableMediaTracking(args: { id: string; options?: RecievedTrackingOptions }) {
+  let conf: TrackingOptions = trackingOptionsParser(args.id, args.options);
+
+  const eventsWithOtherFunctions: Record<string, Function> = {
+    [DocumentEvent.FULLSCREENCHANGE]: (el: HTMLAudioElement | HTMLVideoElement, conf: TrackingOptions) => {
+      if (document.fullscreenElement?.id === args.id) {
+        mediaPlayerEvent(el, DocumentEvent.FULLSCREENCHANGE, conf);
+      }
+    },
+    [MediaEvent.SEEKED]: (el: HTMLAudioElement | HTMLVideoElement, conf: TrackingOptions) => {
+      if (conf.captureEvents.indexOf(SnowplowMediaEvent.PERCENTPROGRESS) !== 0) {
+        while (conf.progress!.boundryTimeoutIds.length) {
+          clearTimeout(Number(conf.progress!.boundryTimeoutIds.pop()));
+        }
+        setPercentageBoundTimeouts(el, conf);
+      }
+    },
+  };
+
+  const eventHandlers: Record<string, Function> = {};
+  for (let ev of conf.captureEvents) {
+    if (eventsWithOtherFunctions.hasOwnProperty(ev)) {
+      eventHandlers[ev] = (el: HTMLAudioElement | HTMLVideoElement) => eventsWithOtherFunctions[ev](el);
+    }
+    eventHandlers[ev] = (el: HTMLAudioElement | HTMLVideoElement, e: MediaEventType) => mediaPlayerEvent(el, e, conf);
+  }
+
+  trackedIds[args.id] = { searchIntervals: [], searchLimit: 5, tracking: false };
+  setUpListeners(args.id, conf, eventHandlers);
+  trackedIds[args.id].searchIntervals.push(setInterval(() => setUpListeners(args.id, conf, eventHandlers), 5000));
+}
+
+function setUpListeners(id: string, conf: TrackingOptions, eventHandlers: Record<string, Function>) {
+  // The element may not be loaded in time for this function to run,
+  // so we have a few goes at finding the element
+  let el = findMediaElem(id);
+
+  if (!trackedIds[id].searchLimit) {
+    console.error("Couldn't find element before timeout");
+    trackedIds[id].searchIntervals.forEach((e: ReturnType<typeof setTimeout>) => clearInterval(e));
+    return;
+  }
+
+  if (!el) {
+    trackedIds[id].searchLimit--;
+    console.error(`Couldn't find element with id: '${id}'. Trying again...`);
```

Comment:
Don't think this is an error, we're going to retry. This error message can be deleted.